CMakeLists.txt

```diff
@@ -12,144 +14,145 @@ option(ENABLE_PARAVIEW_CATALYST
 option(ENABLE_TIOGA "Use TIOGA TPL to perform overset connectivity" OFF)
 option(ENABLE_HYPRE "Use HYPRE Solver library" OFF)
 
-SET(CMAKE_FIND_LIBRARY_PREFIXES lib)
-SET(CMAKE_FIND_LIBRARY_SUFFIXES a)
-set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
-
-MESSAGE("   Trilinos_DIR = ${Trilinos_DIR}")
-SET(CMAKE_PREFIX_PATH ${Trilinos_DIR} ${CMAKE_PREFIX_PATH})
-SET(CMAKE_PREFIX_PATH ${Trilinos_PATH} ${CMAKE_PREFIX_PATH})
+########################## TRILINOS ####################################
+message("   Trilinos_DIR = ${Trilinos_DIR}")
+set(CMAKE_PREFIX_PATH ${Trilinos_DIR} ${CMAKE_PREFIX_PATH})
+set(CMAKE_PREFIX_PATH ${Trilinos_PATH} ${CMAKE_PREFIX_PATH})
 find_package(Trilinos REQUIRED)
+include_directories(${Trilinos_INCLUDE_DIRS})
+include_directories(${Trilinos_TPL_INCLUDE_DIRS})
 
-if (ENABLE_HYPRE)
-  find_package(HYPRE REQUIRED)
+message("\nFound Trilinos!  Here are the details: ")
+message("   Trilinos_DIR = ${Trilinos_DIR}")
+message("   Trilinos_VERSION = ${Trilinos_VERSION}")
+message("   Trilinos_PACKAGE_LIST = ${Trilinos_PACKAGE_LIST}")
+message("   Trilinos_LIBRARIES = ${Trilinos_LIBRARIES}")
+message("   Trilinos_BIN_DIRS = ${Trilinos_BIN_DIRS}")
+message("   Trilinos_INCLUDE_DIRS = ${Trilinos_INCLUDE_DIRS}")
+message("   Trilinos_LIBRARY_DIRS = ${Trilinos_LIBRARY_DIRS}")
+message("   Trilinos_TPL_LIST = ${Trilinos_TPL_LIST}")
+message("   Trilinos_TPL_INCLUDE_DIRS = ${Trilinos_TPL_INCLUDE_DIRS}")
+message("   Trilinos_TPL_LIBRARIES = ${Trilinos_TPL_LIBRARIES}")
+message("   Trilinos_TPL_LIBRARY_DIRS = ${Trilinos_TPL_LIBRARY_DIRS}")
+message("   Trilinos_BUILD_SHARED_LIBS = ${Trilinos_BUILD_SHARED_LIBS}")
+message("   Trilinos_CXX_COMPILER_FLAGS = ${Trilinos_CXX_COMPILER_FLAGS}")
+message("End of Trilinos details\n")
+
+message("Setting and checking of compilers:")
+set(CMAKE_CXX_COMPILER ${Trilinos_CXX_COMPILER} )
+set(CMAKE_C_COMPILER ${Trilinos_C_COMPILER} )
+set(CMAKE_Fortran_COMPILER ${Trilinos_Fortran_COMPILER} )
+# Enable Fortran if it is enabled in Trilinos
+if(CMAKE_Fortran_COMPILER)
```

Comment:
Should this be moved to `project(Nalu CXX Fortran)` ? MasterElements still require the fortran file, so we would require Fortran compiler even if Trilinos is not enabling it? And with `ENABLE_OPENFAST` we definitely need Fortran for `DISCON.f90`

Reply:
Ah, right, I didn't think of that. We probably should move to `project(Nalu CXX Fortran)`. I will do that and do a quick test on it.